### PR TITLE
Snssqs subscription policy

### PR DESCRIPTION
**Pull request by amimimor**
# Description

after an SQS queue is created by daprd, the change introduced modifies the policy it creates to one that limits the SendMessage requests to those only originating from the daprd-created SNS topic, unlike current situation which gives a wildcard ("any") permission to send messages to the aforementioned SQS queue

## Issue reference

Please reference the issue this PR will close: #1230 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by amimimor** [comment]
hi committers. what's happening in terms of (secretstores.azure.keyvault) certification test that are running every 30minutes and report issues. any documentation I can follow along to solve this (if any issue)?

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 .idea
 .vscode
 /vendor
+snssqs_integration_test.go
```
Comment (daixiang0) [inline .gitignore:5]: remove this please.

File: .github/workflows/dapr-bot-schedule.yml [hunk 2]
```diff
@@ -1,61 +0,0 @@
-# ------------------------------------------------------------
```
Comment (daixiang0) [inline .github/workflows/dapr-bot-schedule.yml:1]: please revert this.

**Review by artursouza — APPROVED** [review]

File: pubsub/aws/snssqs/snssqs.go [hunk 3]
```diff
@@ -526,55 +509,89 @@ func (s *snsSqs) createQueueAttributesWithDeadLetters(queueInfo, deadLettersQueu
...
-	// subscribers declare a topic ARN
-	// and declare a SQS queue to use
-	// these should be idempotent
-	// queues should not be created if they exist
+	// subscribers declare a topic ARN and declare a SQS queue to use
+	// these should be idempotent - queues should not be created if they exist.
 	topicArn, err := s.getOrCreateTopic(req.Topic)
 	if err != nil {
-		s.logger.Errorf("error getting topic ARN for %s: %v", req.Topic, err)
+		s.logger.Errorf("error getting topic ARN for %s: %w", req.Topic, err)
```
Comment (CodeMonkeyLeet) [inline pubsub/aws/snssqs/snssqs.go:544]: Wrapped error is not returned

File: pubsub/aws/snssqs/snssqs.go [hunk 4]
```diff
@@ -526,55 +509,89 @@ func (s *snsSqs) createQueueAttributesWithDeadLetters(queueInfo, deadLettersQueu
...
+		s.logger.Errorf("error retrieving SQS queue: %w", err)
 
 		return err
 	}
 
+	// only after a SQS queue and SNS topic had been setup, we restrict the SendMessage action to SNS as sole source
+	// to prevent anyone but SNS to publish message to SQS.
+	err = s.restrictQueuePublishPolicyToOnlySNS(queueInfo, topicArn)
+	if err != nil {
+		s.logger.Errorf("error setting sns-sqs subscription policy: %w", err)
```
Comment (CodeMonkeyLeet) [inline pubsub/aws/snssqs/snssqs.go:562]: Wrapped error is not returned

File: pubsub/aws/snssqs/snssqs.go [hunk 5]
```diff
@@ -526,55 +509,89 @@ func (s *snsSqs) createQueueAttributesWithDeadLetters(queueInfo, deadLettersQueu
...
+	}
+
+	// apply the dead letters queue attributes to the current queue.
 	var deadLettersQueueInfo *sqsQueueInfo
 	if len(s.metadata.sqsDeadLettersQueueName) > 0 {
 		var derr error
 		deadLettersQueueInfo, derr = s.createDeadLettersQueue()
 		if derr != nil {
-			s.logger.Errorf("error creating dead-letter queue: %v", derr)
+			s.logger.Errorf("error creating dead-letter queue: %w", derr)
```
Comment (CodeMonkeyLeet) [inline pubsub/aws/snssqs/snssqs.go:573]: Wrapped error is not returned

File: pubsub/aws/snssqs/snssqs.go [hunk 6]
```diff
@@ -526,55 +509,89 @@ func (s *snsSqs) createQueueAttributesWithDeadLetters(queueInfo, deadLettersQueu
...
 
 			return derr
 		}
 
 		var sqsSetQueueAttributesInput *sqs.SetQueueAttributesInput
 		sqsSetQueueAttributesInput, derr = s.createQueueAttributesWithDeadLetters(queueInfo, deadLettersQueueInfo)
 		if derr != nil {
-			s.logger.Errorf("error creatubg queue attributes for dead-letter queue: %v", derr)
+			s.logger.Errorf("error creatubg queue attributes for dead-letter queue: %w", derr)
```
Comment (CodeMonkeyLeet) [inline pubsub/aws/snssqs/snssqs.go:581]: Wrapped error is not returned

**Review by CodeMonkeyLeet — COMMENTED** [review]
Thanks for the fix and also addressing all the linter issues! Regarding updating the error handling to return wrapped errors, it seems a bit inconsistent, could you take a look?

**Pull request merged**